File: EduqueAPI/app/Models/Cours.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Validation\Rule;

class Cours extends Model
{
    use HasFactory;

    protected $fillable = ['pdf_file', 'pdf_file_name', 'nom_du_cours', 'filiere_id', 'user_id', 'role_user'];
    
    protected $table = 'cours';

    public function filiere()
    {
        return $this->belongsTo(Filiere::class);
    }

    public function user()
    {
       return $this->belongsTo(User::class);
    }

    public function favoris()
    {
        return $this->hasMany(Favori::class);
    }

    public static function validationRules($filiere_id)
    {
        $uniqueRule = Rule::unique('cours')->where(function ($query) use ($filiere_id) {
            return $query->where('filiere_id', $filiere_id);
        });

        return [
            'pdf_file' => [
                'required',
                'mimes:pdf',
                $uniqueRule,
            ],
            'filiere_id' => 'required|exists:filieres,id',
        ];
    }
}

File: education/lib/models/cours.dart
class Cours {
  int id;
  String pdf_file_name;
  String pdf_file;
  String filiere_id;
  final String filiere;
  bool isFavorite;

  Cours({
    required this.id,
    required this.pdf_file_name,
    required this.pdf_file,
    required this.filiere_id,
    required this.filiere,
    this.isFavorite = false,
  });

  factory Cours.fromJson(Map<String, dynamic> json) {
    return Cours(
      id: json['id'],
      pdf_file_name: json['pdf_file_name'],
      filiere_id: json['filiere_id'],
      pdf_file: json['pdf_file'],
      filiere: json['filiere'],
      isFavorite: json['isFavorite'],
      //userId: json['user_id'],
      //roleUser: json['role_user'],
      // Mappez d'autres attributs depuis le JSON au besoin
    );
  }
}
